packages/grid/src/vaadin-grid-selection-column-mixin.js

```diff
@@ -81,12 +81,16 @@ export const GridSelectionColumnMixin = (superClass) =>
     }
 
     /**
-     * Return true if array `a` contains all the items in `b`
+     * Return true if array `left` contains all the items in `right`,
+     * using the `getItemId` method to compare items.
      * We need this when sorting or to preserve selection after filtering.
      * @private
      */
-    __arrayContains(a, b) {
-      return Array.isArray(a) && Array.isArray(b) && b.every((i) => a.includes(i));
+    __arrayContains(left, right) {
```

Comment:
```suggestion
    __selectedItemsInclude(items) {
      return items.every((item) => this.__selectedKeys.has(this.getItemId(item)));
    }
```

Reply:
`__selectedKeys` is on the grid itself, rather than on the selection column. But we could move the method over there to reuse the computed IDs.

Reply:
Moved the method to the selection mixin.